zedd-app/src/AppState.ts

```diff
@@ -539,11 +539,28 @@ export class AppState {
   )
 
   public fillErsatz(when: Interval): void {
-    for (const day of eachDayOfInterval(when)) {
-      this.addSliceIfDayEmpty(
-        this.makeFullDaySlice(day, this.getTaskForName(this.config.ersatzTask)),
-      )
+    this.getHolidays((when.start as Date).getFullYear()).then((holidays) => {
```

Comment:
Passing in the holidays to consider would simplify this function (and keep it sync).